File: easy/1-two-sum/1-two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var dict = [Int: Int]()
        
        for (index, value) in nums.enumerated() {
            if let addent = dict[value] {
                return [addent, index]
            } else {
                dict[target - value] = index
            }
        }
        
        return []
    }
}

File: easy/1-two-sum/1-two-sum.go
func twoSum(nums []int, target int) []int {
	// Create a map to store the complements
	complementMap := make(map[int]int)

	// Iterate through the array
	for i, num := range nums {
		// Calculate the complement
		complement := target - num

		// Check if the complement is in the map
		if index, ok := complementMap[complement]; ok {
			// If the complement is present, return the indices
			return []int{index, i}
		}

		// Store the current element and its index in the map
		complementMap[num] = i
	}

	// If no solution is found, return an empty slice
	return []int{}
}
